Remaining D2k changes to restore original balancing/gameplay

reaperrr opened the pull request:
#9458 + this should wrap up the implementation of @OmegaBolt's PR.

Don't let the number of commits intimidate you, I split everything I could into individual commits for easier reviewing.

reaperrr committed:
Make D2k constr. yard explicit prerequisite

In preparation for moving Upgrades to the player actor in the future.

Additionally remove unnecessary Exit trait from construction yard.

reaperrr committed:
Tweak D2k worm disappear chance, spawn interval and minimum

reaperrr committed:
Disable D2k capturing of husks, make husks explode

reaperrr committed:
Tone down D2k veterancy stat gains

reaperrr committed:
Tweak D2k infantry bounds

reaperrr committed:
Make only light infantry come out of destroyed D2k buildings

reaperrr committed:
Calculate D2k building visibility around center instead of footprint

reaperrr committed:
Reduce range and number of D2k building debris

Fix debris warhead falloff

GraionDilach commented on mods/d2k/rules/defaults.yaml:
```diff
@@ -267,15 +273,16 @@
-		Weapons: Debris, Debris2, Debris3, Debris4
-		Pieces: 3, 7
-		Range: 2c0, 5c0
+		Weapons: debris, debris2, debris3, debris4
```
Is this intended?

reaperrr replied:
No, will fix it.

reaperrr commented:
Updated.

- left veterancy enabled, but toned down stat gains particularly at higher levels
- removed commit that disabled AllowMovement on infantry AutoTarget
- removed commit that disabled some crate actions
- fixed debris typo

OmegaBolt commented:
Why would you keep veterancy and allow movement?

As was discussed in some other commit, veterancy could be purely visual, which I think is the best idea personally so it doesn't interfere with balance and yet gives people that personal attachment to long lasting units.

Allowing movement from autotargeting is one of the absolute most annoying and stupid ideas to put in a game about unit control. The original games never did it, except in TS and RA2 where your anti-aircraft units would chase aircraft and that's commonly considered by the community an annoying bug. I do not want my infantry to ever chase down anything unless I've set it to guard mode, and even then it shouldn't hunt a unit across the whole map.

GraionDilach commented:
Regarding the latter, #9611 would make it mostly redundant.

penev92 commented:
I'm for keeping veterancy. D2k is a mod where your units are dead meat from the moment they exit the factory anyway. Unit preservation is almost non-existent (and that's fine). So I don't think veterancy will have a big (if any) effect on the gameplay (as long as we keep the bonuses small).

pchote commented on mods/d2k/rules/defaults.yaml:
```diff
@@ -267,6 +272,7 @@
 	ShakeOnDeath:
 	Sellable:
 		SellSounds: BUILD1.WAV
+	AutoTargetIgnore:
```
Isn't this going to break the AI?

reaperrr replied:
The AI still attacked my buildings normally in a test game I had.

pchote replied:
Strange.  AFAIK we haven't changed anything relevant to AutoTarget or the AI since the failure from #3337 / #3467.

penev92 replied:
But this *does* break scripted reinforcements in the missions, which can sit idly by my ConYard, chilling.
I think it would be simplest to just not change that here. Can be done in a later PR if you want.

pchote commented on mods/d2k/rules/arrakis.yaml:
```diff
@@ -47,6 +47,7 @@ sandworm:
 		Weapon: WormJaw
 	Sandworm:
 		WanderMoveRadius: 5
+		ChanceToDisappear: 100
```
It would be better to change this in the trait instead, IMO.

pchote commented on mods/d2k/rules/defaults.yaml:
```diff
@@ -124,13 +124,16 @@
-		CaptureThreshold: 1.0
-	TransformOnCapture:
-		ForceHealthPercentage: 25
+#	Capturable:
```
Please don't leave commented out dead code in the rules.

pchote commented on mods/d2k/rules/world.yaml:
```diff
@@ -35,6 +35,10 @@
 		Name: Corrino
 		InternalName: corrino
 		Selectable: false
+	Faction@Smuggler:
```
This doesn't appear to be used yet, but if you want to keep it will need entries set up in the building `FactionImages` dictionaries.

pchote replied:
IIRC smugglers should use ordos(?) artwork for everything except their custom starport, which we currently have set up as corrino.

penev92 replied:
That is correct.

reaperrr replied:
Are smugglers the same as mercenaries? IIRC the AI-controlled faction allied to Ordos in the campaign had a different artwork for their heavy factory, too.

pchote replied:
Nope, mercenaries were yet another faction that copied one of the base factions for everything except their custom heavy factory.

pchote replied:
IMO it would make sense to have a dedicated PR for setting up the smuggers/mercenaries(/fremen?) factions properly.

penev92 replied:
Smugglers used Ordos units and structures and had their own Starport.
Mercenaries used Ordos units and structures and had their own Heavy Factory.
Fremen used Atreides units and structures.
House Corrino (the Imperial family) used Harkonnen units and structures and had their own Palace.

Might as well leave those for a separate PR.

MustaphaTR replied:
According to Tibed in orginal game Corrino has own CYard and War Factory. But they are same as Harkonnen ones as graphics and code. Possibly used in missions to separate them from Harkonnen ones.

pchote commented on mods/d2k/rules/world.yaml:
```diff
@@ -59,6 +63,8 @@ World:
 		ProductionPaletteWidget: PRODUCTION_PALETTE
 	WormManager:
+		Minimum: 0
+		SpawnInterval: 6000
```
The defaults could be changed instead.

pchote commented on mods/d2k/rules/structures.yaml:
```diff
@@ -59,8 +60,9 @@ construction_yard:
 	Production:
-		Produces: Building, Upgrade
+		Produces: Building
 	Exit:
```
I don't think we need this anymore, so this trait can be removed (it was only used for upgrades, which have now moved to the player).

pchote commented on mods/d2k/rules/defaults.yaml:
```diff
@@ -222,7 +225,7 @@
 	AppearsOnRadar:
 		UseLocation: yes
-	HiddenUnderFog:
+	HiddenUnderShroud:
```
This will make aircraft visible to *everyone* under the fog.

reaperrr replied:
IIRC the main reason @OmegaBolt did this was that Carryalls and Ornithopters would otherwise be much harder to shoot down than in the original.

pchote replied:
I assumed that it was done to make them visible to their owner under the fog.  That should be done by removing the `AlwaysVisibleStances` line below instead.

reaperrr commented:
Updated.

Not sure about the aircraft HiddenUnderFog, should I leave that in or remove it?

reaperrr commented:
Removed the aircraft visibility change. I guess we should let people play some games, if lots of them complain we can still change it.

penev92 commented:
- [ ] Please include mission `atreides 02b` to the first commit as well.
 - [ ] I am now able to build any structure apart from concrete slabs without a ConYard.

abcdefg30 commented on mods/d2k/maps/atreides-01a/map.yaml:
```diff
@@ -145,8 +147,8 @@ Rules:
-		Production:
-			Produces: Building
+		Buildable:
+			Prerequisites: ~disabled
```
Shouldn't `construction_yard` be unabuildable by default?
If so, please remove this (also from the other missions).

reaperrr commented:
@penev92 @abcdefg30 fixed prerequisites and missions.

pchote commented:
Changes look ok, and didn't notice anything that looked too bogus while spectating a bot match with these changes, so :+1:.

penev92 commented:
Adding "Changes requested" for building `AutoTargetIgnore` breaking scripted reinforcements.

reaperrr commented:
Updated.

- removed AutoTargetIngore commit
- removed warhead falloff changes from debris commit, this is properly taken care of in my warhead PR now

penev92 commented on mods/d2k/maps/atreides-01a/map.yaml:
```diff
@@ -134,6 +134,8 @@ Rules:
 		MissionObjectives:
 			EarlyGameOver: true
+		Production:
+			Produces: Building
```
Is this still necessary? Selling my ConYard has an odd side-effect.
While that's not an issue, you need to just exclude Upgrades here, not move Building to the player, right?

penev92 commented on mods/d2k/rules/arrakis.yaml:
```diff
@@ -32,11 +32,11 @@ sandworm:
 		Description: Attracted by vibrations in the sand.\nWill eat units whole and has a large appetite.
 	Health:
 		HP: 1200
```
Partially related: it takes 2 trikes to kill a worm before it can eat one of them.
TibEd says Sandworm has 9990 HP, so this definitely needs a buff.

reaperrr commented:
Updated.

- Fixed Production issue
- Increased worm HP, but increased target rescan delay from 32 to 125 ticks and reduced search radius from 27 to 20 cells

Mailaender commented on mods/d2k/rules/structures.yaml:
```diff
@@ -895,9 +895,6 @@ conyard.harkonnen:
 	Buildable:
-		Queue: Building
-		BuildPaletteOrder: 1000
-		Prerequisites: ~disabled
```
I guess this makes the construction yard appear twice with cheats on:

![image](https://cloud.githubusercontent.com/assets/756669/10716777/b4c3c88c-7b43-11e5-8df6-bcd0e2acb53d.png)

reaperrr commented:
Updated, fixed the cyard double appearance.

penev92 commented:
Adding "Changes requested" for the first commit as discussed on IRC.

reaperrr commented:
Updated.

obrakmann commented:
I'll relieve this one of its misery as well. :+1:

obrakmann commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/9107ff42356ced738edf4a126be541fc8d9b2eaa)
